### ISSUE TITLE
Additional commands for save/restore permissions for migration (team A to team B)

### ISSUE THREAD

**Issue by watermint** [watermint/toolbox#59]
# command

```
team migration preserve -path PATH_TO_PRESERVED_STATE
team migration transfer -from FROM_TEAM_ALIAS -to TO_TEAM_ALIAS -path PATH_TO_PRESERVED_STATE
```

# process

## preserve

* save team member list
* save groups and members
* save namespace list
* save namespace member list

## transfer

* add admin user into all shared folders (FROM)
* mirror team folders (FROM to TO)
* detach users (FROM)
* invite users (TO)
* create groups (TO)
* restore sharing permissions (TO)
    * team folder
    * shared folder
    * group
* remove permissions from all shared folders of admin user (TO)

**Comment by watermint** [comment]
Split migration process into below three

1. check ... preflight check (verify status, etc)
2. content ... copy contents to dest team
3. transfer ... transfer accounts & permissions

**Comment by watermint** [comment]
- [ ] Add option archive team folder upon completion (mirror + verification)

**Comment by watermint** [comment]
Can't preserve permissions if archived on content mirror step

**Comment by watermint** [comment]
- [ ] Create option for resume process from preserved state

**Comment by watermint** [comment]
- [ ] Group members of context must be refactored as `map[string]map[string]*mo_group_member.Member` (groupId -> email)
- [ ] Namespace members of context must be refactored as `map[string]map[string]*mo_...Member`

**Comment by watermint** [comment]
- [ ] Shared folder owner must be transferred to admin during Bridge stage (because members are not accepted invitation)

```
ERROR	Unable to add member to shared folder	{"name": "UCTMSF-Prj-Team", "folderId": "xxxxxxxxxx", "member": "xxx+xxxx@xxxxxxxxx.xxx", "error": "Endpoint specific error[access_error] access_error/not_a_member/"}
```

**Comment by watermint** [comment]
- [ ] Verify available licenses of dest team

**Comment by watermint** [comment]
https://github.com/watermint/toolbox/pull/94

**Comment by watermint** [comment]
Shipped as part of release 45.2
https://github.com/watermint/toolbox/pull/97
